File: psets/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
    /* getting user input */
    string text = get_string("Text: ");
    
    /* things that are needed */
    int words, letters, sentences, i;
    words = letters = sentences = i = 0;

    /* calculations */
    while (text[i] != '\0')
    {
        /* if the character is an alphhabet */
        if (((text[i] >= 'a') && (text[i] <= 'z')) || ((text[i] >= 'A') && (text[i] <= 'Z')))
        {
            letters = letters + 1;
        }

        /* if the chararcter is a space */
        else if (text[i] == ' ')
        {
            words = words + 1;
        }

        /* if the character is a full-stop, exclamation or a question mark */
        else if ((text[i] == '?') || (text[i] == '.') || (text[i] == '!'))
        {
            sentences = sentences + 1;
        }

        // else do nothing

        /* incrementing the looping variable */
        i = i + 1;
    }

    /**
     * the last space is not recorded but it ends the sentence
     * and also the word
    */
    words = words + 1;

    /**
     *  doing the math
     *  L = number of letters per 100 words
     *  S = number of sentences per 100 words
    */
    float L = ((100.0 / words) * letters);
    float S = ((100.0 / words) * sentences);

    /* The formula */
    int grade = round((0.0588 * L) - (0.296 * S) - 15.8);

    /* the grading begins */
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    /* philosopher cadre */
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    /* plebean class */
    else
    {
        printf("Grade %i\n", grade);
    }

    /* tradition demands ... */
    return 0;
}

File: pset6/readability/readability.py
text: str = input("Text: ")

# the alphabets
alpha: str = 'abcdefghijklmnopqrstuvwxyz'
letters: int = 0
words: int = 1
sentences: int = 0
for char in text:
    if (char in alpha) or (char in alpha.upper()):
        letters += 1
    elif char in (' ',):
        words += 1
    elif char in ('!', '.', '?'):
        sentences += 1
    else:
        continue

L: float = 100 * letters / words
S: float = 100 * sentences / words

index: int = round(0.0588 * L - 0.296 * S - 15.8)
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
